stdlib/REPL/src/REPL.jl

```diff
@@ -1239,4 +1242,14 @@ function start_repl_server(port::Int)
     end
 end
 
+print_exit_hint(repl::BasicREPL, line) = print_exit_hint(repl.terminal, line)
+print_exit_hint(repl::LineEditREPL, line) = print_exit_hint(repl.t, line)
+print_exit_hint(repl::StreamREPL, line) = print_exit_hint(repl.stream, line)
+function print_exit_hint(io::IO, line)
+    if lowercase(strip(line)) == "exit"
```

Comment:
Would it make sense to also print this if the input is "quit" (that's what e.g. GAP uses) ?

Anyway, I really like this PR, it fixes one of those tiny papercuts which seem irrelevant in isolation but add up.

Reply:
I wouldn't object to also doing this for "quit", but I'll defer to @JeffBezanson.

I'd want to keep this limited to a very small set of inputs. For example, you can exit R by typing `q()` and pressing enter. But I definitely wouldn't want to print a hint if someone types `q` in the Julia REPL.

I think it makes sense for `exit` because the corresponding function `exit()` is a function that actually exists in Julia. But we don't actually have a `quit` function in Julia.

Reply:
But I don't feel strongly either way. Happy to add `quit`, also happy to leave it out.

Reply:
Yes I think we should add it.

Reply:
I like what the python repl does:
```
>>> quit
Use quit() or Ctrl-D (i.e. EOF) to exit
>>> quit = 2
>>> quit
2
```
It stops printing the message if `quit` is defined as a variable. Stylish! I think we should do this too.
Can be checked with `Base.isbindingresolved(Main, :exit) && isdefined(Main, :exit)`.

Reply:
In python, that's not magic, it is just literally the definition of show for that default variable:
```
>>> exit.__class__
<class 'site.Quitter'>
>>> quit.__class__
<class 'site.Quitter'>
```

The equivalent here would be:
```
julia> using Base: exit as quit; # @eval Base const quit = exit; @eval Base export quit;

julia> Base.show(io::IO, ::MIME"text/plain", ::typeof(exit)) = print(io, "Use exit() or Ctrl-D (i.e. EOF) to quit")

julia> repr(exit)
"exit"

julia> show(exit)
exit
julia> summary(exit)
"typeof(exit)"

julia> display(exit)
Use exit() or Ctrl-D (i.e. EOF) to quit

julia> exit
Use exit() or Ctrl-D (i.e. EOF) to quit

julia> exit()
```

Reply:
Which would have been https://github.com/JuliaLang/julia/pull/21362